### PR TITLE
Ml on gcp encoder decoder

### PR DESCRIPTION
**Pull request by BenoitDherin**

### CODE REVIEW

File: courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb [hunk 1]
```diff
@@ -0,0 +1,1058 @@
+{
```
Comment (munnm) [inline courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>'taken' --&gt; 'token'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (munnm) [inline courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>RRN --&gt; RNN</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (munnm) [inline courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>'encoder-decoder decoder' --&gt; 'encoder-decoder'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (munnm) [inline courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>'encoder_sate' --&gt; 'encoder_state'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (munnm) [inline courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>trian --&gt; train</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/778/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

**Comment by munnm** [comment]
<p>hi Benoit. The notebook looks great! I only saw very minor typos worth fixing. Sorry for the delay.</p>

File: courses/machine_learning/deepdive2/text_classification/solutions/utils_preproc.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+"""Utilities for preprocessing natural language for machine learning"""
+import re
+import unicodedata
+
+import tensorflow as tf
+
+
+def unicode_to_ascii(s):
+    normalized = unicodedata.normalize('NFD', s)
```
Comment (munnm) [inline courses/machine_learning/deepdive2/text_classification/solutions/utils_preproc.py]: should there be doc strings as in the other functions?

File: courses/machine_learning/deepdive2/text_classification/solutions/utils_preproc.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    return ''.join(c for c in normalized if unicodedata.category(c) != 'Mn')
+
+
+def preprocess_sentence(w):
+    w = unicode_to_ascii(w.lower().strip())
```
Comment (munnm) [inline courses/machine_learning/deepdive2/text_classification/solutions/utils_preproc.py]: docstring? prob not necessary but nice to have.

File: courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb (hunk 1 above)
Comment (BenoitDherin) [reply courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>Done</p>

Comment (BenoitDherin) [reply courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>Done</p>

Comment (BenoitDherin) [reply courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>Done</p>

Comment (BenoitDherin) [reply courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>Done</p>

Comment (BenoitDherin) [reply courses/machine_learning/deepdive2/text_classification/solutions/rnn_encoder_decoder.ipynb]: <p>Done</p>

**Comment by BenoitDherin** [comment]
<p>Thanks for the review!</p>

**Pull request merged**